app/src/main/java/com/duckduckgo/app/tabs/ui/TabSwitcherActivity.kt

```diff
@@ -440,18 +454,18 @@ class TabSwitcherActivity : DuckDuckGoActivity(), TabSwitcherListener, Coroutine
 
             val popupBinding = PopupTabsMenuBinding.bind(popupMenu.contentView)
             val viewState = viewModel.selectionViewState.value
-            val numSelectedTabs = (viewModel.selectionViewState.value.mode as? Selection)?.selectedTabs?.size ?: 0
 
-            layoutTypeMenuItem = menu.createDynamicInterface(numSelectedTabs, popupBinding, binding.tabsFab, toolbar, viewState.dynamicInterface)
+            val numSelectedTabs = viewModel.selectionViewState.value.numSelectedTabs
+            menu.createDynamicInterface(numSelectedTabs, popupBinding, binding.tabsFab, toolbar, viewState.dynamicInterface)
         } else {
             menuInflater.inflate(R.menu.menu_tab_switcher_activity, menu)
             layoutTypeMenuItem = menu.findItem(R.id.layoutTypeMenuItem)
-        }
 
-        when (viewModel.layoutType.value) {
-            LayoutType.GRID -> showListLayoutButton()
-            LayoutType.LIST -> showGridLayoutButton()
-            null -> layoutTypeMenuItem?.isVisible = false
+            when (viewModel.layoutType.value) {
```

Comment:
The layout button state is now updated in `menu.createDynamicInterface()` when multi-selection is enabled.